app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -2,14 +2,16 @@
 
 class OmniauthCallbacksController < ApplicationController
   def login_dot_gov
-    auth = request.env['omniauth.auth']
-    @user = User.from_omniauth(auth)
-    return unless @user.persisted?
+    @user = User.from_omniauth(request.env['omniauth.auth'])
 
-    reset_session
-    set_user_session
+    if @user.persisted? && @user.approval_status != 'not_approved'
```

Comment:
As this is adding logic for a `not_approved` user, there should be a corresponding spec with the context `when a user is not_approved`.